src/mdacli/save.py

```diff
@@ -197,6 +155,194 @@ def stack_1d_arrays_list(list_1D, extra_list=None):
+            save_result_array(value, fprefix=fprefix, arr_name=key)
+            keys.append(key)
+
+    return return_with_remove(results, keys, remove)
+
+
+def is_array_higher_dim(arr):
+    """Assert if arr is array of more than one dim."""
+    valid = \
+        isinstance(arr, np.ndarray) \
+        and arr.ndim > 1
+
+    return valid
+
+
+def save_result_array(arr, fprefix='prefix', arr_name='arr_name'):
+    """Save array to disk accoring to num of dimensions."""
+    # Remove extra dimensions
+    item = np.squeeze(arr)
+    n_dims = item.ndim
+
+    if n_dims == 2:
+        savetxt_w_command(
+            fname=f"{fprefix}_{arr_name}.csv",
+            X=item,
+            )
+
+    elif n_dims == 3:
+        save_3D_array_to_2D_csv(
+            item,
+            arr_name=f"{fprefix}_{arr_name}",
+            zipit=True,
+            )
+
+    elif n_dims > 3:
+        np.save(
+            f"{fprefix}_{arr_name}.npz",
+            item,
+            allow_pickle=False,
+            )
+
+    else:
+        warnings.warn(
+            Emphasise.warning(
+                f"Could not save the array if {n_dims} dimensions. "
+                "Something unexpected happend."
+                )
+            )
+
+    return
+
+
+def save_3D_array_to_2D_csv(
+        item,
+        arr_name='arr',
+        zipit=True,
+        ):
+    """
+    Save 3D array to 2D CSVs.
+
+    Has option to store all in a ZIP file.
+    """
+    min_dim = np.argmin(item)
+    files_to_zip = []
+
+    # Split array along the dimension with smallest number
+    # of entries
+    splitted_item = np.split(
+        item,
+        item.shape[min_dim],
+        axis=min_dim,
+        )
+
+    save_to_folder = not zipit
+    folder = None
+    if save_to_folder:
+        folder = Path(arr_name)
+        folder.mkdir(parents=True, exist_ok=True)
+
+    for i, arr in enumerate(splitted_item):
+        fname = f"{arr_name}_dim_{min_dim}_idx_{i}.csv"
+        foutname = folder.joinpath(fname) if folder else fname
+        savetxt_w_command(fname=foutname, X=np.squeeze(arr))
+        files_to_zip.append(fname)
+
+    if zipit:
+        save_files_to_zip(files_to_zip, zipname=arr_name, remove=True)
+
+    return
+
+
+def save_json_serializables(results, remove=True, **jsonargs):
+    """Save serializable items to a JSON."""
+    json_dict = {
+        key: value
+        for key, value in results.items()
+        if is_serializable(value)
+        }
+
+    if remove:
+        for key in json_dict.keys():
+            results.pop(key)
+
+    if json_dict:
+        save_to_json(json_dict, **jsonargs)
+
+
+def is_serializable(value):
+    """Assert if value is json serializable."""
+    try:
+        json.dumps(value)
+        return True
+    except (TypeError, OverflowError):
+        return False
+
+
+def save_to_json(json_dict, fname='jdict', indent=4, sort_keys=True):
+    """Save dictionary to JSON file."""
+    with open(f'{fname}.json', 'w') as f:
```

Comment:
Add a header here. Something like
```python
f.write(f"//{get_cli_input()\n}")
```

Reply:
You can't add headers to JSON. If you do that it will cease to be a JSON file and will be just a text file difficult to parse.
We can add a key to the dictionary to store the command as a string though :-)

Reply:
Ahh I see. A key is also nice.

Reply:
done on last commit